engine/apps/slack/tasks.py

```diff
@@ -580,6 +613,20 @@ def populate_slack_channels_for_team(slack_team_identity_id):
         SlackChannel.objects.bulk_update(
             channels_to_update, fields=("name", "is_archived", "is_shared", "last_populated"), batch_size=5000
         )
+        if rate_limited:
+            # save collected channels ids to cache and restart the task with the current pagination cursor
+            cache.set(collected_channels_key, collected_channels, timeout=3600)
+            delay = random.randint(1, 3) * 60
+            logger.warning(
+                f"'conversations.list' slack api error: rate_limited. SlackTeamIdentity pk: {slack_team_identity_id}. "
+                f"Delay populate_slack_channels_for_team task for {delay//60} min."
+            )
+            start_populate_slack_channels_for_team(slack_team_identity_id, delay, cursor)
+        else:
+            # delete excess channels
```

Comment:
Consider adding `assert collected_channels` just before deleting. It might be redundant, but this way it's harder to accidentally wipe out all channels when making changes to this task.